iptstate.cc

```diff
@@ -715,6 +716,7 @@ void get_input(WINDOW *win, string &input, const string &prompt,
       // 8 is shift-backspace - just incase
       case KEY_BACKSPACE:
       case 8:
+      case 127:
```

Comment:
huh? 127 isn't - afaik - a standard keycode at all. This ... has to be some very, very bizarre terminal emulator.

Reply:
I'm using Apple Terminal app. There is an option in the preferences to send `^H` (8) for delete, but it's unchecked by default so it sends char 127 (DEL). Is DEL also not a valid alternative to delete characters?

Reply:
Ah if you want the delete key, use `KEY_DC` - that seems to work for me across a variety of terminal emulators.

As for what it's sending, according to `xev`, `DEL` is 119

```
KeyPress event, serial 36, synthetic NO, window 0x6a00001,
    root 0x291, subw 0x6a00002, time 590607549, (23,11), root:(668,77),
    state 0x10, keycode 119 (keysym 0xffff, Delete), same_screen YES,
    XLookupString gives 1 bytes: (7f) ""
    XmbLookupString gives 1 bytes: (7f) ""
    XFilterEvent returns: False
```

Though on my keyboard and with rxvt terminal, ncurses gets `330` from the DELETE key... :/

Reply:
On my setup (with Apple Terminal app) backspace is coming through in Ubuntu 16.04 and Arch 4.17.12 as char 127:
```
$ showkey -a
Press any keys - Ctrl-D will terminate this program
^? 	127 0177 0x7f
```
On Ubuntu this is somehow magically working with iptstate (maybe older ncurses lib?) but on Arch not.

Here is a link that argues that backspace could be either char 8 or 127:
http://www.ibb.net/~anne/keyboard.html

Reply:
On my debian box, I also see backspace as 127, but it works. But `ncurses`, should handle this stuff magically. Meh. I'm fine with this if it helps people. I'll do an inline comment above cause I"d like to fix up some comments here.